### DISCUSSION TITLE
在安卓平台如何使用Hardwarebuffer提交纹理？

### DISCUSSION THREAD

**Discussion by xzpqjew** [Tencent/tgfx#66] [General] [answered]
我在TGFX的开源公告中，看到了TGFX说支持hardwarebuffer纹理提交在安卓平台，请问在安卓平台，如果用mediacodec解码视频，如何从其config的surface上，拿到解码的纹理缓存对应的hardwarebuffer，传给tgfx渲染，有没有hardwarebuffer使用例子

**Comment by domchen** [comment, marked as answer]
可以参考 libpag 里的用法 ：https://github.com/Tencent/libpag/blob/main/src/platform/android/HardwareDecoder.cpp 主要的代码就是包装 tgfx 的 SurfaceTextureReader 类。视频流用刚刚那个方法。你如果只有一帧的 HardwareBuffer 图像。直接用：
 ```
tgfx::Image::MakeFrom(HardwareBufferRef hardwareBuffer, YUVColorSpace colorSpace = YUVColorSpace::BT601_LIMITED);
```
就可以绘制了。
